src/psyclone/tests/psyGen_test.py

```diff
@@ -4531,18 +4719,44 @@ def test_fparser2astprocessor_handling_unaryopbase(f2008_parser):
     from fparser.common.readfortran import FortranStringReader
     from fparser.two.Fortran2003 import Execution_Part, UnaryOpBase
     reader = FortranStringReader("x=-4")
-    fparser2unary_operation = Execution_Part.match(reader)[0][0].items[2]
-    assert isinstance(fparser2unary_operation, UnaryOpBase)
+    fp2unaryop = Execution_Part.match(reader)[0][0].items[2]
+    assert isinstance(fp2unaryop, UnaryOpBase)
 
     fake_parent = Node()
     processor = Fparser2ASTProcessor()
-    processor.process_nodes(fake_parent, [fparser2unary_operation], None)
+    processor.process_nodes(fake_parent, [fp2unaryop], None)
     # Check a new node was generated and connected to parent
     assert len(fake_parent.children) == 1
     new_node = fake_parent.children[0]
     assert isinstance(new_node, UnaryOperation)
     assert len(new_node.children) == 1
-    assert new_node._operator == '-'
+    assert new_node._operator == UnaryOperation.Operator.MINUS
+
+    # Test parsing all supported unary operators.
+    testlist = (('+', UnaryOperation.Operator.PLUS),
+                ('-', UnaryOperation.Operator.MINUS),
+                ('.not.', UnaryOperation.Operator.NOT),
+                ('.NOT.', UnaryOperation.Operator.NOT),
+                )
+
+    for opstring, expected in testlist:
+        fake_parent = Node()
+        fp2unaryop.items = (opstring, fp2unaryop.items[1])
```

Comment:
Again, please add a comment describing what you're doing.

Reply:
Added comment